src/Akeneo/Tool/Bundle/ElasticsearchBundle/Client.php

```diff
@@ -118,6 +119,14 @@ public function bulkIndexes($documents, $keyAsId = null, Refresh $refresh = null
             'items' => [],
         ];
         foreach ($documents as $document) {
+            $extraActions = [];
+            $documentObject = $document;
+            if ($document instanceof AffectedByMigrationProjection) {
+                $document = $document->toArray();
+                if ($documentObject->shouldBeMigrated()) {
+                    $extraActions[] = ['delete' => ['_index' => $this->indexName, '_id' => $documentObject->getFormerDocumentId()]];
+                }
+            }
```

Comment:
```suggestion
            if ($document instanceof AffectedByMigrationProjection) {
                if ($document->shouldBeMigrated()) {
                    $extraActions[] = ['delete' => ['_index' => $this->indexName, '_id' => $document->getFormerDocumentId()]];
                }
                $document = $document->toArray();
            }
```

In order to remove the weird `$documentObject`